vesting/src/lib.rs

```diff
@@ -22,6 +22,9 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
+// Disable the following two lints since they originate from an external macro (namely decl_storage)
+#![allow(clippy::redundant_closure_call, clippy::string_lit_as_bytes)]
+
```

Comment:
@thiolliere Is this something we can fix in Substrate?  Or is it best for people building on Substrate to suppress the warning like this?

Reply:
It is used to suppress warnings similar to this
```
warning: calling `as_bytes()` on a string literal
  --> pallets/adding-machine/src/lib.rs:17:1
   |
17 | / decl_storage! {
18 | |     trait Store for Module<T: Trait> as AddingMachine {
19 | |         Sum get(fn sum): u32;
20 | |     }
21 | | }
   | |_^
   |
   = note: `#[warn(clippy::string_lit_as_bytes)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#string_lit_as_bytes
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
I think this is the cause https://github.com/paritytech/substrate/blob/ea501c2d61d27e7f0f76f93e5fa7e5c90882da2f/frame/support/procedural/src/storage/storage_struct.rs#L109

Reply:
yes we have fixed some in the past I'll take a look to fix those